src/Http/Http.Extensions/gen/Microsoft.AspNetCore.Http.ValidationsGenerator/Parsers/ValidationsGenerator.TypesParser.cs

```diff
@@ -25,10 +25,22 @@ internal ImmutableArray<ValidatableType> ExtractValidatableTypes(IInvocationOper
         var parameters = operation.TryGetRouteHandlerMethod(operation.SemanticModel, out var method)
             ? method.Parameters
             : [];
+
+        var fromServiceMetadataSymbol = wellKnownTypes.Get(
+            WellKnownTypeData.WellKnownType.Microsoft_AspNetCore_Http_Metadata_IFromServiceMetadata);
+
         var validatableTypes = new HashSet<ValidatableType>(ValidatableTypeComparer.Instance);
         List<ITypeSymbol> visitedTypes = [];
+
         foreach (var parameter in parameters)
         {
+            // Skip attributes that implement the IFromServiceMetadata interface.
+            // These attributes are used for dependency injection (DI) purposes and do not require validation.
+            if (parameter.GetAttributes().Any(attr => attr.AttributeClass is not null && attr.AttributeClass.ImplementsInterface(fromServiceMetadataSymbol)))
```

Comment:
Should also add a check for `[FromKeyedServicesAttribute]` (and test of course)

Reply:
Yep -- added a test to check for the `[FromKeyedServices]`.